src/schema/v2/artworksForUser/helpers.ts

```diff
@@ -1,21 +1,46 @@
 import { ResolverContext } from "types/graphql"
 import { CursorPageable } from "relay-cursor-paging"
-import { extractNodes } from "lib/helpers"
+import { convertConnectionArgsToGravityArgs, extractNodes } from "lib/helpers"
 import gql from "lib/gql"
 
-export const getNewForYouRecs = async (
+export const getNewForYouArtworkIDs = async (
   args: CursorPageable,
   context: ResolverContext
 ): Promise<string[]> => {
   const {
     authenticatedLoaders: {
       vortexGraphqlLoader: vortexGraphQLAuthenticatedLoader,
+      auctionLotRecommendationsLoader: auctionLotRecommendationsAuthenticatedLoader,
     },
     unauthenticatedLoaders: {
       vortexGraphqlLoader: vortexGraphQLUnauthenticatedLoader,
+      auctionLotRecommendationsLoader: auctionLotRecommendationsUnauthenticatedLoader,
     },
     xImpersonateUserID,
   } = context
+
+  if (args.includeAuctionArtworks) {
+    if (!auctionLotRecommendationsAuthenticatedLoader)
+      throw new Error("Authentication failed: User is not authenticated.")
+
+    const userID = args.userId || xImpersonateUserID
+    const gravityArgs = convertConnectionArgsToGravityArgs(args)
+    const { page, size } = gravityArgs
+
+    // When a `userID` is specified, this is an app request and we should use the
+    // unauthenticated loader. The authenticated loader is for logged-in users.
+    const loader = userID
+      ? auctionLotRecommendationsUnauthenticatedLoader
+      : auctionLotRecommendationsAuthenticatedLoader
+
+    const { data } = await loader({
```

Comment:
Any/every fetch and loader call can throw :}

Personally I think it's fine to not handle those errors, as those get handled by the default error handler. The main reason to do a `catch` is if you want to customize the error message, or rather than actually `throw` you want to handle the error and return a `[]`, or something like that. Especially if the field is non-nullable, you might want to return `0`, `[]`, etc.

In this case, the field returns a connection (and is nullable), so I don't think any specific error-handling needs to happen.

In the event this call fails, it's probably fine to return a `null` / and include the `errors` (vs. partially fulfilling and letting the backfill return the full set in that case).